src/Microsoft.ML.LightGBM/LightGbmRankingTrainer.cs

```diff
@@ -50,7 +50,8 @@ private static VersionInfo GetVersionInfo()
         protected override uint VerCategoricalSplitSerialized => 0x00010005;
         public override PredictionKind PredictionKind => PredictionKind.Ranking;
 
-        public LightGbmRankingModelParameters(IHostEnvironment env, TreeEnsemble trainedEnsemble, int featureCount, string innerArgs)
+        [BestFriend]
+        internal LightGbmRankingModelParameters(IHostEnvironment env, TreeEnsemble trainedEnsemble, int featureCount, string innerArgs)
```

Comment:
Hi Wei-Sheng, it is not clear to me that outside code is using this, could you confirm? If it is not I might prefer to make this just `internal` and not add the `[BestFriend]` attribute. I have similar comments on other things that were made `internal` while also being `[BestFriend]`ed.

Reply:
Sorry Tom. I supposed to pushing a re-doing iteration to hide those things before you seeing them... but got blocked by hiding more and more staff. This should look much better hopefully later today.

---
In reply to: [251181975](https://github.com/dotnet/machinelearning/pull/2243#discussion_r251181975) [](ancestors = 251181975)